src/lib/client.ts

```diff
@@ -183,14 +183,12 @@ export class Client {
     return signedBlock.block.header.number.toNumber()
   }
 
-  /* SEGMENT INDEX */
-  // Used to check archived segments locally and display farmer plot status.
-  public async getLocalFarmerSegmentIndex(): Promise<number> {
+  public async getLocalSegmentCount(): Promise<number> {
     const plot_progress_tracker =
       ((await invoke("plot_progress_tracker")) as number) / 256
```

Comment:
This will work, but there are a few issues:
* `plot_progress_tracker` is a bad name, should be `get_plotted_pieces_count()` or something like that
* `256` will work for now, but it is not guaranteed to be that way in the future, the actual value must be extracted from the runtime: https://github.com/subspace/subspace/blob/main/crates/sc-consensus-subspace/src/lib.rs#L913
* as the result of above and considering that you should already be connected to local node anyway I suggest the following:
    * subscribe to events of local node and monitor for root block archiving that contains archived segment in it, making UI reactive rather than relying on timers
    * remove `plot_progress_tracker` on backend

Non-blocking, can be done in a separate PR, just don't forget about this.

cc @ozgunozerk

Reply:
Thought some more about this. If we care about plotting as such, we do have to inspect the plot and not the chain, but I'm not sure there is a major difference between those right now until we implement DSN sync of the farmer.

Reply:
I've opened the issue #133 for tracking this